src/Workspaces/Remote/Core/SolutionAssetStorage.Scope.cs

```diff
@@ -18,12 +18,12 @@ internal partial class SolutionAssetStorage
     internal sealed partial class Scope(
         SolutionAssetStorage storage,
         Checksum solutionChecksum,
-        SolutionState solution) : IDisposable
+        SolutionCompilationState solution) : IDisposable
     {
         private readonly SolutionAssetStorage _storage = storage;
 
         public readonly Checksum SolutionChecksum = solutionChecksum;
-        public readonly SolutionState Solution = solution;
+        public readonly SolutionCompilationState Solution = solution;
```

Comment:
```suggestion
        public readonly SolutionCompilationState CompilationState = compilationState;
```

(but can be a follow-up if that's got a lot of impact)

Reply:
tracked in followup